sonic_platform_base/sensor_fs.py

```diff
@@ -0,0 +1,159 @@
+"""
+    sensor_fs.py
+
+    File system based sensors implementation. 
+"""
+
+from sonic_platform_base.sensor_base import SensorBase
+from sonic_platform_base.sensor_base import VoltageSensorBase
+from sonic_platform_base.sensor_base import CurrentSensorBase
+import logging
+
+
+class SensorFs(SensorBase):
+    """Implementation of file system based sensor class"""
+
+    __defaults = {
+        'name': '',
+        'sensor': '',
+        'high_thresholds': ['N/A', 'N/A', 'N/A'],
+        'low_thresholds': ['N/A', 'N/A', 'N/A'],
+        'position': -1,
+    }
+
+    def __init__(self, sensor_type='sensor', **kw):
+
+        super(SensorFs, self).__init__()
+
+        if 'sensor' not in kw:
+            raise Exception('Failed to initialize sensor')
+
+        for k,v in self.__defaults.items():
+            setattr(self, k, kw.get(k,v))
+
+        if (len(self.high_thresholds) != 3 or len(self.low_thresholds) != 3):
+            raise Exception('{}: Missing sensor thresholds'.format(self.name))
+
+        self.minimum_sensor = self.get_value()
+        self.maximum_sensor = self.minimum_sensor
+
+    def get_name(self):
+        """Returns the sensor name"""
+        return self.name
+
+    def get_value(self):
+        """Returns the sensor measurement"""
+        try:
+            with open(self.sensor) as f:
+                return int(f.readline().rstrip())
+        except:
+            return None
+
+    def get_high_threshold(self):
+        """Returns the sensor high threshold value"""
+        return self.high_thresholds[1]
+
+    def get_low_threshold(self):
+        """Returns the sensor low threshold value"""
+        return self.low_thresholds[1]
+
+    def set_high_threshold(self, value):
```

Comment:
I didn't go through all the related PRs, maybe I missed them, but I want to understand here, will the new threshold value be saved? if not if the daemon restarts how it will get the previous set value?

Reply:
This is simply providing an implementation for the base class. The threshold values in default case are picked from the sensors.yaml file. So on daemon restart, the file is read again and they are populated. This is currently not used but is provided for any future use.